### PR TITLE
Show deleted messages in channel preview + missing channel item redesign

### PR DESCRIPTION
**Pull request by nuno-vieira**
### 🔗 Issue Links

[IOS-1506](https://linear.app/stream/issue/IOS-1506/v5-qa-deleted-message-should-be-present-in-channel-preview)

Depends on: https://github.com/GetStream/stream-chat-swift/pull/4024

### 🎯 Goal

Deleted messages should be visible in the channel list preview.

### 📝 Summary

- Compute the channel preview message in the UI layer from `channel.latestMessages`, filtering only ephemeral messages
- Show deleted messages in the channel preview with nosign icon and "Message deleted" text
- Remove `InjectedChannelInfo` and simplify `ChatChannelListItem` with inline private computed properties
- Remove `@MainActor` extensions on `ChatChannel` — formatting logic now lives in the view
- Update "No messages" to "No messages yet" per Figma
- Add snapshot tests for deleted, pending, ephemeral-skipped, and empty message previews
- Add unit tests for `MessagePreviewFormatter` handling of deleted messages

### 🛠 Implementation

The `ChatChannel.previewMessage` property was removed from the core SDK (see linked PR). The preview message is now computed in the SwiftUI view layer as `channel.latestMessages.first(where: { $0.type != .ephemeral })`, which allows deleted messages to appear in the preview while still filtering out ephemeral (Giphy) messages.

`InjectedChannelInfo` was removed as it was unnecessary indirection — its only real purpose was suppressing unread badges for the selected channel, which is now handled by a simple `isSelected: Bool` property.

### 🧪 Manual Testing Notes

- Open a channel and send a few messages, then delete the latest — verify the preview shows "Message deleted" with nosign icon
- For deleted messages you sent, verify "You:" prefix appears with no read indicators
- Send a message in a poor connection scenario — verify the clock icon appears for pending state

### ☑️ Contributor Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] This change should be manually QAed
- [x] Changelog is updated with client-facing changes
- [ ] Changelog is updated with new localization keys
- [x] New code is covered by unit tests
- [ ] Documentation has been updated in the `docs-content` repo

## Summary by CodeRabbit

* **New Features**
  * Channel list shows a “Message deleted” placeholder (includes “You:” when from the current user).

* **Updates**
  * Message previews skip ephemeral/pending items and show the previous visible message.
  * Read/delivery indicators, draft text and timestamp/subtitle now reflect the selected previewed message.
  * Unread badge visibility depends on channel selection.
  * Empty-channel text changed to “No messages yet.”

### CODE REVIEW

File: DemoAppSwiftUI/PinChannelHelpers.swift [hunk 1]
```diff
@@ -70,6 +70,38 @@ struct DemoAppChatChannelListItem: View {
...
+    }
+
+    private var subtitleText: String {
+        if shouldShowTypingIndicator {
+            return channel.typingIndicatorString(currentUserId: chatClient.currentUserId)
+        }
+        if let previewMessage {
+            return utils.messagePreviewFormatter.format(previewMessage, in: channel)
+        }
+        return "No messages yet"
```
Comment (martinmitrevski) [inline DemoAppSwiftUI/PinChannelHelpers.swift:102]: we should localize this text - probably it's always been like this

File: DemoAppSwiftUI/PinChannelHelpers.swift [hunk 2]
```diff
@@ -15,7 +15,7 @@ struct DemoAppChatChannelListItem: View {
 
     var channel: ChatChannel
     var channelName: String
-    var injectedChannelInfo: InjectedChannelInfo?
+    var isSelected: Bool = false
```
Comment (martinmitrevski) [inline DemoAppSwiftUI/PinChannelHelpers.swift:18]: what does it mean, that the channel is opened?

File: Sources/StreamChatSwiftUI/ChatChannelList/ChatChannelListViewModel.swift [hunk 3]
```diff
@@ -545,11 +545,7 @@ import UIKit
         if skippedChannelUpdates && selectedChannel == nil {
```
Comment (martinmitrevski) [inline Sources/StreamChatSwiftUI/ChatChannelList/ChatChannelListViewModel.swift]: we can now merge these 2 if statements though, right?

File: Sources/StreamChatSwiftUI/ChatChannelList/ChatChannelListViewModel.swift [hunk 4]
```diff
@@ -562,30 +558,6 @@ import UIKit
             }
         }
     }
-
-    private func updateSelectedChannelData() {
```
Comment (martinmitrevski) [inline Sources/StreamChatSwiftUI/ChatChannelList/ChatChannelListViewModel.swift:566]: This was crazy. However, I believe we should test before iOS 16, when a navigation view is used. I think the screen was popped in some cases, which was why we had this stuff.

**Review by martinmitrevski — COMMENTED** [review]
Looks good, will do some tests next

File: DemoAppSwiftUI/PinChannelHelpers.swift (hunk 1 above)
Comment (nuno-vieira) [reply DemoAppSwiftUI/PinChannelHelpers.swift:102]: This is Demo App only though 🤔

File: DemoAppSwiftUI/PinChannelHelpers.swift (hunk 2 above)
Comment (nuno-vieira) [reply DemoAppSwiftUI/PinChannelHelpers.swift:18]: Yes exactly

File: Sources/StreamChatSwiftUI/ChatChannelList/ChatChannelListViewModel.swift (hunk 3 above)
Comment (nuno-vieira) [reply Sources/StreamChatSwiftUI/ChatChannelList/ChatChannelListViewModel.swift]: What do you mean? 🤔

File: Sources/StreamChatSwiftUI/ChatChannelList/ChatChannelListViewModel.swift (hunk 4 above)
Comment (nuno-vieira) [reply Sources/StreamChatSwiftUI/ChatChannelList/ChatChannelListViewModel.swift:566]: OK, @testableapple reminder to test this, I don't have a sim with this version, but if you don't have it as well, I'll install

Comment (martinmitrevski) [reply Sources/StreamChatSwiftUI/ChatChannelList/ChatChannelListViewModel.swift:566]: actually, what you can do instead is temporarily comment out the navigation stack code in NavigationContainerView and test with navigation view. You should force channel update from a channel that's not selected (just sending a channel in another channel should be fine).

File: DemoAppSwiftUI/PinChannelHelpers.swift (hunk 1 above)
Comment (martinmitrevski) [reply DemoAppSwiftUI/PinChannelHelpers.swift:102]: ah ok, I didn't notice

File: Sources/StreamChatSwiftUI/ChatChannelList/ChatChannelListViewModel.swift (hunk 3 above)
Comment (martinmitrevski) [reply Sources/StreamChatSwiftUI/ChatChannelList/ChatChannelListViewModel.swift]: When you see it like this, you will know 😄  
```swift
private func handleChannelAppearance() {
        if skippedChannelUpdates && selectedChannel == nil {
            updateChannels()
        } else if skippedChannelUpdates {
            updateChannels()
        }
    }
```

Comment (nuno-vieira) [reply Sources/StreamChatSwiftUI/ChatChannelList/ChatChannelListViewModel.swift]: Ah shiit XD, will fix this

File: Sources/StreamChatSwiftUI/ChatChannelList/ChatChannelListViewModel.swift (hunk 4 above)
Comment (nuno-vieira) [reply Sources/StreamChatSwiftUI/ChatChannelList/ChatChannelListViewModel.swift:566]: I tested, and it all looks good 👍 (But maybe still worth testing on iOS 16)

**Review by martinmitrevski — APPROVED** [review]
LGTM ✅  The e2e checks are failing though.
